.env.example

```diff
@@ -45,10 +45,7 @@ RUN_BULLMQ_WORKERS=runs,runs-cleanup,vectorStores-cleanup,vectorStores-fileProce
 # --- BACKEND SECTION ---
 
 # LLM backend, possible values are: ollama, openai, watsonx, bam, ibm-vllm
-LLM_BACKEND=watsonx
-
-# Embedding backend, possible values are: ollama, openai, watsonx, bam, caikit
-EMBEDDING_BACKEND=watsonx
+AI_BACKEND=watsonx
```

Comment:
I unified backends under one variable and refactored the factories to be in one place.

Having separate EMBEDDING and LLM providers is a rare use-case and confusing to setup. We can keep it separate if necessary, let me know what you think :)